File: 13_NumberLineJumps/numberLineJumps.py
def numberLineJumps(x1: int, v1: int, x2: int, v2: int) -> str:
    return "YES" if v1>v2 and (x2-x1)%(v1-v2)==0 else "NO"

if __name__ == '__main__':
    kangaroos = input("Insert the data: ").rstrip().split()

    x1 = int(kangaroos[0])
    v1 = int(kangaroos[1])

    x2 = int(kangaroos[2])
    v2 = int(kangaroos[3])

    result = numberLineJumps(x1, v1, x2, v2)

    print(result)

File: 13_NumberLineJumps/NumberLineJumps.java
import java.util.Scanner;

import java.util.List;
import java.util.stream.Stream;
import java.util.stream.Collectors;

public class NumberLineJumps {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        System.out.print("Insert the data: ");
        String[] kangaroos = in.nextLine().replaceAll("\\s+$", "").replaceAll("\\s+", " ").split(" ");

        int x1 = Integer.parseInt(kangaroos[0]);
        int v1 = Integer.parseInt(kangaroos[1]);

        int x2 = Integer.parseInt(kangaroos[2]);
        int v2 = Integer.parseInt(kangaroos[3]);

        String result = Result.numberLineJumps(x1, v1, x2, v2);

        System.out.println(result);

        in.close();
    }
}

class Result {
    public static String numberLineJumps(int x1, int v1, int x2, int v2) {
        return (v1>v2 && (x2-x1)%(v1-v2)==0) ? "YES" : "NO";
    }
}

